implicit conversion of nullptr constant to 'bool'

### What happened?

build with default options will trigger the following errors.  
(force  add -DIREE_ENABLE_WERROR_FLAG=OFF  with cmake will resolve the break temporarily)

```
[77/1446] Building CXX object compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/LinkTuningSpecsPass.cpp.o
FAILED: compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/LinkTuningSpecsPass.cpp.o 
ccache /usr/bin/clang++ -DIREE_ALLOCATOR_SYSTEM_CTL=iree_allocator_libc_ctl -DIREE_HAVE_LIBBACKTRACE=1 -I/home//workspace/experimental/iree -I/home//workspace/experimental/iree-build -I/home//workspace/experimental/iree/third_party/llvm-project/llvm/include -I/home//workspace/experimental/iree-build/llvm-project/include -I/home//workspace/experimental/iree/third_party/llvm-project/mlir/include -I/home//workspace/experimental/iree-build/llvm-project/tools/mlir/include -I/home//workspace/experimental/iree/third_party/llvm-project/lld/include -I/home//workspace/experimental/iree-build/llvm-project/tools/lld/include -I/home//workspace/experimental/iree/compiler/src -I/home//workspace/experimental/iree-build/compiler/src -I/home//workspace/experimental/iree/runtime/src -I/home//workspace/experimental/iree-build/runtime/src -I/home//workspace/experimental/iree-build/runtime/src/iree/base/internal/flatcc -I/home//workspace/experimental/iree/llvm-external-projects/iree-dialects/include -I/home//workspace/experimental/iree-build/llvm-external-projects/mlir-iree-dialects/include -I/home//workspace/experimental/iree/compiler/bindings/c/. -I/home//workspace/experimental/iree-build/compiler/bindings/c/. -isystem /home//workspace/experimental/iree-build/_deps/libbacktrace_src-src -isystem /home//workspace/experimental/iree/third_party/flatcc/include -O2 -g   -gsplit-dwarf -ggnu-pubnames -fPIC -fvisibility=hidden -fno-rtti -fno-exceptions -Werror -Wall -Wno-error=deprecated-declarations -Wno-extern-c-compat -Wno-invalid-offsetof -Wno-unused-const-variable -Wno-unused-function -Wno-unused-private-field -Wctad-maybe-unsupported -Wfloat-overflow-conversion -Wfloat-zero-conversion -Wfor-loop-analysis -Wformat-security -Wgnu-redeclared-enum -Wimplicit-fallthrough -Winfinite-recursion -Wliteral-conversion -Wlogical-op-parentheses -Wnon-virtual-dtor -Woverloaded-virtual -Wpointer-arith -Wself-assign -Wstring-conversion -Wtautological-overlap-compare -Wthread-safety -Wthread-safety-beta -Wunused-comparison -Wvla -fno-omit-frame-pointer -fmacro-prefix-map=/home//workspace/experimental/iree=iree -DIREE_THREADING_ENABLE=1 -pthread -I/home//workspace/experimental/iree/third_party/flatcc/include/ -I/home//workspace/experimental/iree/third_party/flatcc/include/flatcc/reflection/ -std=gnu++17 -MD -MT compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/LinkTuningSpecsPass.cpp.o -MF compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/LinkTuningSpecsPass.cpp.o.d -o compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/LinkTuningSpecsPass.cpp.o -c /home//workspace/experimental/iree/compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp
/home//workspace/experimental/iree/compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp:426:28: error: implicit conversion of nullptr constant to 'bool' [-Werror,-Wnull-conversion]
      /* restrictRoot = */ nullptr, /* flattenResults = */ nullptr,
                           ^~~~~~~
                           false
/home//workspace/experimental/iree/compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp:426:60: error: implicit conversion of nullptr constant to 'bool' [-Werror,-Wnull-conversion]
      /* restrictRoot = */ nullptr, /* flattenResults = */ nullptr,
                                                           ^~~~~~~
                                                           false
2 errors generated.
```

### Steps to reproduce your issue

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


### What component(s) does this issue relate to?

_No response_

### Version information

_No response_

### Additional context

_No response_

--- a/compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp
+++ b/compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp
@@ -409,12 +409,8 @@ static FailureOr<NamedSequenceOp> emitLinkedDefaultTuningSpec(ModuleOp module) {
   SmallVector<Attribute> mergedActions;
 
   for (ForeachMatchOp foreachMatchOp : foreachMatchOps) {
-    ArrayAttr matchers = foreachMatchOp.getMatchers();
-    ArrayAttr actions = foreachMatchOp.getActions();
-    for (auto [matcher, action] : llvm::zip_equal(matchers, actions)) {
-      mergedMatchers.push_back(cast<SymbolRefAttr>(matcher));
-      mergedActions.push_back(cast<SymbolRefAttr>(action));
-    }
+    llvm::append_range(mergedMatchers, foreachMatchOp.getMatchers());
+    llvm::append_range(mergedActions, foreachMatchOp.getActions());
   }
 
   Region &region = newEntryPoint.getRegion();
@@ -423,8 +419,8 @@ static FailureOr<NamedSequenceOp> emitLinkedDefaultTuningSpec(ModuleOp module) {
   builder.setInsertionPointToStart(body);
   auto mergedForeachMatch = ForeachMatchOp::create(
       builder, loc, resultTypes, newEntryPoint.getArgument(0),
-      /* forwarded_inputs = */ ValueRange(),
-      /* restrictRoot = */ nullptr, /* flattenResults = */ nullptr,
+      /*forwarded_inputs=*/ValueRange(),
+      /*restrict_root=*/false, /*flatten_results=*/false,
       builder.getArrayAttr(mergedMatchers),
       builder.getArrayAttr(mergedActions));
   transform::YieldOp::create(builder, loc, mergedForeachMatch->getResult(0));

